requirements_tools/check_requirements.py

```diff
@@ -321,12 +334,15 @@ def _expected_pinned(filename, pin_filename):
                 'Is it missing from {}?\n'
                 '\t- {}\n'.format(filename, pin_filename, req.key),
             )
-        ret.add('{}=={}'.format(req.key, installed_things[req.key].version))
+        ret.add(f'{req.key}=={installed_things[req.key].version}')
         ret |= get_pinned_versions_from_requirement(req)
+    if ret:
+        assert isinstance(ret, set) and isinstance(list(ret)[0], str)
```

Comment:
argh, sorry! I missed this while working on the type checking - will fix this tomorrow